### ISSUE TITLE
Feature request: cache passthrough

### ISSUE THREAD

**Issue by kjyv** [saracen/lfscache#11]
Hey,
Your tool is quite nice, especially since downloading from AWS is ridiculously slow for us lately.

It would be very useful in our local network with multiple machines using multiple github repositories if we could just have a static dns entry from e.g. github-cloud.s3.amazonaws.com to the local lfscache machine. That way all personal access tokens will stay valid, laptops moving to other networks don't have to change their repositories lfs path everytime etc.

For that it would be necessary to be able to configure git-lfs to use the full original URL as with github.com, so /org/repo.git/info/lfs which would be used by lfscache instead of the --url parameter (the machine with lfscache would have to not use the local dns and then access the same HTTP Location URL it got from the request). Additionally, since it would only be necessary to cache certain repos, supplying a list of URLs for which the current cache is used would be useful and then all other requests would simply be passed through.

So, not sure if this is something you would want to add and if it's too much work, but let me know what you think.

**Comment by saracen** [comment]
@kjyv if I understand correctly, this would mean regular traffic to github.com (say, from a browser), would also be proxied over `lfscache`, it just wouldn't act on it?

I think this would be better setup as a reverse proxy within your organization. For example, you could configure nginx as a reverse proxy, with a rule where path `/org/repo.git/info/lfs` instead uses lfscache.

**Comment by kjyv** [comment]
I agree a reverse proxy for the regular github.com access is better but the issue then remains that for every repository you want to have a cache for, the url has to be configured separately. We probably won't need it anymore, regular github connections work fast again.
